{% include 'code/templates/header.html' %}

      <ul id="info" class="bam" data-url='__EGA_MAIN_URL__/files/{{ stable_id }}'>
	<!-- Content table -->
	<li data-title="Inferred assembly">{{ report.InferredAssembly }}</li>
	<li data-title="Total reads">{{ report.TotalReads|numberFormat(' ') }}</li>
	<li data-title="Read length">{{ report.ReadLength }}</li>
	<li data-title="Insert size">{{ report.InsertSize|numberFormat(' ') }}</li>
	<li data-title="Average base quality">{{ report.AvgBaseQuality }}</li>
	<li data-title="Average base coverage">{{ report.AvgBaseCoverage }}</li>
	<li data-title="Number of samples">{{ report.NbSamples }}</li>
	<li data-title="Encoding information">{{ report.EncodingInfo }}</li>
	<!-- File info table -->
      </ul>

      <h1>File Data</h1>

      <div id="data">
	<section id="base-coverage" class="wide" data-ifmissing="Missing data">
	  <h2>Base Coverage Distribution</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>This chart represents the base coverage distribution
	       along the reference file. Y-axis represents the number
	       of times a position in the reference file is
	       covered. The x-axis represents the range of the values
	       for the coverage.
	    </p>
	    <p>Data is represented in a log scale to minimise the
	       variability. A high peak in the beginning (low
	       coverage) and a curve descending is expected.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="base-quality" class="wide" data-ifmissing="Missing data">
	  <h2>Base Quality</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>The base quality distribution shows the Phred quality
	       scores describing the probability that a nucleotide has
	       been incorrectly assigned; e.g. an error in the
	       sequencing. Specifically, Q=-log<sub>10</sub>(P),
	       where <em>Q</em> is the Phred score and <em>P</em> is
	       the probability the nucleotide is wrong. The larger the
	       score, the more confident we are in the base
	       call. Depending on the sequencing technology, we can
	       expect to see different distributions, but we expect to
	       see a distribution skewed towards larger (more
	       confident) scores; typically around 40.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="mapped-reads">
	  <h2>Mapped Reads</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>Number of reads successfully mapped (singletons & both
	       mates) to the reference genome in the sample. Genetic
	       variation, in particular structural variants, ensure
	       that every sequenced sample is genetically different
	       from the reference genome it was aligned to. Small
	       differences against the reference are accepted, but,
	       for more significant variation, the read can fail to be
	       placed. Therefore, it is not expected that the mapped
	       reads rate will hit 100%, but it is supposed to be high
	       (usually &gt;90%). Calculations are made taking into
	       account the proportion of mapped reads against the
	       total number of reads (mapped/mapped+unmapped).
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="both-mates-mapped">
	  <h2>Both Mates Mapped</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>When working with paired-end sequencing, each DNA
	       fragment is sequenced from both ends, creating two
	       mates for each pair. This chart shows the fraction of
	       reads in pairs where both of the mates successfully map
	       to the reference genome.  .</p>
	    <p>Notice that reads not mapped to the expected distance
	       are also included as occurs with the proper pairs chart.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="singletons">
	  <h2>Singletons</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>When working with paired-end sequencing, each DNA
	       fragment is sequenced from both ends, creating two
	       mates for each pair. If one mate in the pair
	       successfully maps to the reference genome, but the
	       other is unmapped, the mapped mate is a singleton. One
	       way in which a singleton could occur would be if the
	       sample has a large insertion compared with the
	       reference genome; one mate can fall in sequence
	       flanking the insertion and will be mapped, but the
	       other falls in the inserted sequence and so cannot map
	       to the reference genome. There are unlikely to many
	       such structural variants in the sample, or sequencing
	       errors that would cause a read not to be able to
	       map. Consequently, the singleton rate is expected to be
	       very low (&lt;1%).
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="forward-strand">
	  <h2>Forward Strand</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>Fraction of reads mapped to the forward DNA strand. The
	       general expectation is that the DNA library preparation
	       step will generate DNA from the forward and reverse
	       strands in equal amounts so after mapping the reads to
	       the reference genome, approximately 50% of them will
	       consequently map to the forward strand. Deviations from
	       the 50%, may be due to problems with the library
	       preparation step.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="proper-pairs">
	  <h2>Proper Pairs</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>A fragment consisting of two mates is called a proper
	       pair if both mates map to the reference genome at the
	       expected distance according to the reference genome. In
	       particular, if the DNA library consists of fragments
	       ~500 base pairs in length, and 100 base pair reads are
	       sequenced from either end, the expectation would be
	       that the two reads map to the reference genome
	       separated by ~300 base pairs. If the sequenced sample
	       contains large structural variants, e.g. a large
	       insertion, where we expect the reads mapping with a
	       large separation would be a signal for this variant,
	       and the reads would not be considered as proper
	       pairs. Based on the sequencing technology, there is
	       also an expectation of the orientation of each read in
	       the fragment.
	    </p>
	    <p>The rate of proper pairs is expected to be well over
	      90%; even if the mapping rate itself is low as a result
	      of bacterial contamination, for example.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="duplicates">
	  <h2>Duplicates</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>PCR duplicates are two (or more) reads that originate
	       from the same DNA fragment. When sequencing data is
	       analyzed, it is assumed that each observation
	       (i.e. each read) is independent; an assumption that
	       fails in the presence of duplicate reads. Typically,
	       algorithms look for reads that map to the same genomic
	       coordinate, and whose mates also map to identical
	       genomic coordinates. It is important to note that as
	       the sequencing depth increases, more reads are sampled
	       from the DNA library, and consequently it is
	       increasingly likely that duplicate reads will be
	       sampled. As a result, the true duplicate rate is not
	       independent of the depth, and they should both be
	       considered when looking at the duplicate
	       rate. Additionally, as the sequencing depth in
	       increases, it is also increasingly likely that reads
	       will map to the same location and be marked as
	       duplicates, even when they are not. As such, as the
	       sequencing depth approaches and surpasses the read
	       length, the duplicate rate starts to become less
	       indicative of problems.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="mapping-quality" class="wide" data-ifmissing="Missing data">
	  <h2>Mapping Quality Distribution</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>The mapping quality distribution shows the Phred
	       quality scores describing the probability that a read
	       does not map to the location that it has been assigned
	       to (specifically, Q=-log<sub>10</sub>(P),
	       where <em>Q</em> is the Phred score and <em>P</em> is
	       the probability the read is in the wrong location). So
	       the larger the score, the higher the quality of the
	       mapping. Some scores have a specific meaning, e.g. a
	       score of 0 means that the read could map equally to
	       multiple places in the reference genome. The majority
	       of reads should be well mapped, and so we expect to see
	       this distribution heavily skewed to a significant value
	       (typically around 60). It is not unusual to see some
	       scores around zero. Reads originating from repetitive
	       elements in the genome will plausibly map to multiple
	       locations.
	    </p>
	  </div>
	  <svg></svg>
	</section>
	<section id="mapped-unmapped" class="wide" data-ifmissing="Missing data">
	  <h2>Mapped <em>vs</em> Unmapped</h2>
	  <span class="trigger"></span>
	  <div>
	    <p>Stacked column chart for both mapped and unmapped reads
	       along all chromosomes in the reference file. It is a
	       similar representation as shown in the Mapped reads
	       chart but for each chromosome. Although sequenced
	       sample may be a female, it is possible to get reads in
	       the Y chromosome as there are common regions in both
	       chromosomes called pseudoautosomal regions (PAR1,
	       PAR2).
	    </p>
	    <p>Unmapped reads belonging to each chromosome are
	       determined when the one mate/pair is aligned and the
	       other is not. The unmapped read should have chromosome
	       and POS identical to its mate. It could also be due
	       when aligning is performed with bwa as it concatenates
	       all the reference sequences together, so if a read
	       hangs off of one reference onto another, it will be
	       given the right chromosome and position, but it also be
	       classified as unmapped.
	    </p>
	  </div>
	  <svg></svg>
	</section>
      </div>


    </main>
    <footer>
      <p id="copyright">
        &copy; Copyright {{ year }}. EGA Consortium
      </p>
      
      <p id="inspired-by">Data extracted with <a href="https://github.com/samtools/samtools"> samtools</a>.</p>
      <p id="plots-by">The stats plots are rendered using
	the <a href="https://github.com/deweylab/RSEM/blob/master/samtools-1.3/misc/plot-bamstats">plot-bamstats
	script</a> according to
	the <a href="http://www.htslib.org/doc/samtools.html">samtools
	reference manual</a>.</p>
      
      <p id="funded-by">
        Funded by <img src="{{ static('img/eu-flag.jpg') }}" alt="European Union" />
      </p>
    </footer>

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
    <script>window.jQuery || document.write('<script src="{{ static('js/jquery.min.js') }}">\x3C/script>')</script>
    <script src="https://d3js.org/d3.v5.min.js"></script>
    <script>window.d3 || document.write('<script src="{{ static('js/d3.v5.min.js') }}">\x3C/script>')</script>
    
    <script src="{{ static('js/egafile-utils.js') }}"></script>
    <script src="{{ static('js/egafile-pie.js') }}"></script>
    <script src="{{ static('js/egafile-curve.js') }}"></script>
    <script src="{{ static('js/egafile-bars.js') }}"></script>
    <script src="{{ static('js/egafile-stackbars.js') }}"></script>
    <script>
      window.ega_pie(document.querySelector("#mapped-reads svg"), {{ report.Data.MappedReads|safe }});
      window.ega_pie(document.querySelector("#forward-strand svg"), {{ report.Data.ForwardStrand|safe }});
      window.ega_pie(document.querySelector("#proper-pairs svg"), {{ report.Data.ProperPair|safe }});
      window.ega_pie(document.querySelector("#singletons svg"), {{ report.Data.Singletons|safe }});
      window.ega_pie(document.querySelector("#both-mates-mapped svg"), {{ report.Data.BothMatesMapped|safe }});
      window.ega_pie(document.querySelector("#duplicates svg"), {{ report.Data.Duplicates|safe }});
      window.ega_curve("#base-coverage svg",
                      JSON.parse("{{ report.Data.BaseCoverageDistribution|escapejs }}"),
                      { 'xAxis': 'Coverage value',
                        'yAxis': '# Bases'});
      window.ega_bars("#base-quality svg",
                      JSON.parse("{{ report.Data.BaseQualityDistribution|escapejs }}"),
                      { 'xAxis': 'Phred quality score',
                        'yAxis': '# Bases',
                        'yAxisWidth': 50 });
      window.ega_bars("#mapping-quality svg",
                      JSON.parse("{{ report.Data.MappingQualityDistribution|escapejs }}"),
                      { 'xAxis': 'Phred quality score',
                        'yAxis': '# Reads'});
      window.ega_stacked_bars("#mapped-unmapped svg", JSON.parse("{{ report.Data.MappedUnmapped|escapejs }}"));
    </script>
  </body>
</html>
